### DISCUSSION TITLE
Route Authentication in Boot store undefined when using Pinia?

### DISCUSSION THREAD

**Discussion by donPuerto** [quasarframework/quasar#9268] [General - Components / Directives / etc] [answered]
Hi I'm using Pinia and try the Route Authentication, However store is undefined?

``` js
export default boot(({ router, store }) => {
  router.beforeEach((to, from, next) => {
    // Now you need to add your authentication logic here, like calling an API endpoint

    console.log('store', store)  // undefined
 
    next()
  })
})
```

Here's my Pinia setup in Boot also
```js
import { boot } from 'quasar/wrappers';
import { createPinia } from 'pinia'

const pinia = createPinia()
export default boot(({app}) => {

    app.use(pinia)
    console.log('Pinia was created', pinia)
    app.config.globalProperties.$pinia = pinia
});

export { pinia }
```

**Comment by ontwikkelfabriek** [comment]
Just tried your router boot code/file and it consoled the store object just fine.

Could you post the output of 'quasar info' ?

**Comment by stefanvanherwijnen** [comment]
You probably did not select Vuex (store) when creating your project with Quasar CLI. If I remember correctly Quasar checks for a store folder when bundling and if it exists it adds `store` to the boot files.
I would suggest setting up a new project and try if it works.

**Reply by donPuerto** [reply]
Oh okay, Thanks

**Comment by rstoenescu** [comment, marked as answer]
The `store` is supplied only when using the Vuex one (so you have the /src/store folder).
In your case, you are creating Pinia in a boot file and correctly exporting the store. So wherever you need it, just import it from the boot file. Example: `import { pinia } from 'boot/<your-pinia-boot-file>'`. Just make sure that the pinia boot file is placed to run before all others that import from it.

**Reply by donPuerto** [reply]
Can we have out-of-the-box choices for installation (Pinia and Vuex)? Pinia is much better than Vuex 4 in many ways  ;)
Vuex 5 still on a pipe
